### DISCUSSION TITLE
[Writing Gentoo ebuild] Minimum version of cargo-make required to build?

### DISCUSSION THREAD

**Discussion by Interpause** [filips123/PWAsForFirefox#295] [General]
Hello, I am trying to write a Gentoo ebuild to build the native component of PWAsForFirefox. This is actually my first time writing an ebuild, but if it goes well, I might try to PR it to GURU so that it is available for Gentoo users.

Anyways, the ebuild guide suggests versioning dependencies. While most can be auto-extracted from `Cargo.lock`, `cargo-make` isn't included in it. Could I know if there is a minimum version of `cargo-make` required? Thanks.

**Comment by filips123** [comment]
I don't know which minimum version is required, but I guess it can be pretty much any version, unless they changed something major in the past.

**Reply by Interpause** [reply]
Took me longer than expected but the extension is able to detect so is now a matter of is there any bugs caused by the ebuild & testing if the ebuild works properly on other systems.
https://github.com/Interpause/sh-scripts/blob/b3b76040f8c4aa216a3bc437c9dba9abe7b02c85/ebuilds/www-plugins/firefoxpwa-2.4.1.ebuild

**Reply by filips123** [reply]
Thanks! I'll probably check Gentoo documentation because I'm not really familiar with it, but at first glance it seems fine.

Also, is it possible to determine and set those dependencies and hashes automatically? If possible and allowed by Gentoo guidelines, I could maybe automatically publish a new version to their repos using GitHub Actions when a new release is created, so users could have newest versions as soon as possible.

**Reply by Interpause** [reply]
The dependencies were auto-detected from `Cargo.lock` using https://github.com/gentoo/cargo-ebuild. But it seems the tool currently cannot auto-detect crates installed via `git`, so unfortunately https://github.com/Interpause/sh-scripts/blob/main/ebuilds/www-plugins/firefoxpwa-2.4.1.ebuild#L301-L306 is manual.

I'm actually a relatively inexperienced Gentoo user, but I think the process of getting a new ebuild into the main Gentoo tree is actually fairly involved. It might be easier to setup automatic publishing to the community tree instead (https://wiki.gentoo.org/wiki/Project:GURU).

**Reply by Interpause** [reply]
btw thanks for the plugin. so far it has been working fine and some of the features surpass Chromium PWAs lol

**Comment by Interpause** [comment]
Updated ebuild to 2.5.0: https://github.com/Interpause/sh-scripts/blob/main/ebuilds/www-plugins/firefoxpwa-2.5.0.ebuild

Also, decided to try and see if I can get the ebuild added to Gentoo GURU: https://bugs.gentoo.org/904071

**Comment by Interpause** [comment]
Just pushed the ebuild to GURU, now a matter of time for the commit to be reviewed and merged to main.
https://gitweb.gentoo.org/repo/proj/guru.git/commit/?h=dev&id=03c0dfbf39dead18dcb756569acc2fb93ff981bd

**Reply by filips123** [reply]
Thanks. When `cargo-ebuild` gets support for Git dependencies, I'll check if I can automate updating on releases.

**Comment by filips123** [comment]
Changes to `rust-url` were merged to the upstream repo, so the package now uses it instead of my fork. You'll probably have to change this in the package for 2.7.0.

This also means that those changes will probably be released to crates.io soon:tm:. (However, `rust-url` seems to be returning to maintenance after some time of inactivity, so it might still take some time until they finish all things they want before release.) Once this is released, I'll also check if I can get rid of or publish the remaining two/three Git crates.

Also, I've found that [pycargoebuild](https://github.com/projg2/pycargoebuild) apparently supports crates from Git. I haven't fully tested it yet, because it needs to be run on Gentoo, but I will maybe check in the future if it can be used to automate publishing.
